go/cmd/dolt/commands/dump.go

```diff
@@ -0,0 +1,291 @@
+
+	var fileName string
+	resultFormat, _ := apr.GetValue(formatParam)
+
+	switch resultFormat {
+	case "", "sql", ".sql":
+		fileName = "doltdump.sql"
+	case "csv", ".csv":
+		//handle CSV filetype
+		//maybe create dir 'doltdump' and put all the csv dump files into it
+	default:
+		usage()
+		return nil, errhand.BuildDError("invalid result format").Build()
+	}
+
+	//fileLoc := getExportDestination(apr.Arg(0))
+	fileLoc := getExportDestination(fileName)
+
+	return &exportOptions{
+		//tableName:   tableName,
+		force:  apr.Contains(forceParam),
+		format: resultFormat,
+		//src:         tableLoc,
+		dest: fileLoc,
+	}, nil
+}
+
+type DumpCmd struct{}
+
+// Name is returns the name of the Dolt cli command. This is what is used on the command line to invoke the command
+func (cmd DumpCmd) Name() string {
+	return "dump"
+}
+
+// Description returns a description of the command
+func (cmd DumpCmd) Description() string {
+	return "Export all tables in the working set into a file."
+}
+
+// CreateMarkdown creates a markdown file containing the help text for the command at the given path
+func (cmd DumpCmd) CreateMarkdown(wr io.Writer, commandStr string) error {
+	ap := cmd.createArgParser()
+	return CreateMarkdown(wr, cli.GetCommandDocumentation(commandStr, lsDocs, ap))
+}
+
+func (cmd DumpCmd) createArgParser() *argparser.ArgParser {
+	ap := argparser.NewArgParser()
+	//ap.ArgListHelp = append(ap.ArgListHelp, [2]string{"file", "The file being output to."})
+	ap.SupportsFlag(forceParam, "f", "If data already exists in the destination, the force flag will allow the target to be overwritten.")
+	ap.SupportsString(formatParam, "r", "result_file_type", "Define the type of the output file. Valid values are sql and csv. Defaults to sql.")
+
+	return ap
+}
+
+func importStatsCB(stats types.AppliedEditStats) {
+	noEffect := stats.NonExistentDeletes + stats.SameVal
+	total := noEffect + stats.Modifications + stats.Additions
+	displayStr := fmt.Sprintf("Rows Processed: %d, Additions: %d, Modifications: %d, Had No Effect: %d", total, stats.Additions, stats.Modifications, noEffect)
+	displayStrLen = cli.DeleteAndPrint(displayStrLen, displayStr)
+}
+
+// EventType returns the type of the event to log
+func (cmd DumpCmd) EventType() eventsapi.ClientEventType {
+	return eventsapi.ClientEventType_DUMP
+}
+
+// Exec executes the command
+func (cmd DumpCmd) Exec(ctx context.Context, commandStr string, args []string, dEnv *env.DoltEnv) int {
+	ap := cmd.createArgParser()
+	_, usage := cli.HelpAndUsagePrinters(cli.GetCommandDocumentation(commandStr, dumpDocs, ap))
+
+	root, verr := GetWorkingWithVErr(dEnv)
+	if verr != nil {
+		return HandleVErrAndExitCode(verr, usage)
+	}
+
+	tblNames, err := doltdb.GetNonSystemTableNames(ctx, root)
+	if err != nil {
+		errhand.BuildDError("error: failed to get tables").AddCause(err).Build()
+	}
+
+	if len(tblNames) == 0 {
+		cli.Println("No tables to export.")
+		return 0
+	}
+
+	exOpts, verr := parseExportArgs(ap, commandStr, args)
```

Comment:
Call this getExportOpts (not parsing) and move the business logic out of the returned struct